pkg/controllers/monitoring-stack/components.go

```diff
@@ -475,6 +521,31 @@ func newPrometheus(
 					},
 				},
 			},
+			Affinity: &corev1.Affinity{
+				PodAntiAffinity: &corev1.PodAntiAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+						{
+							TopologyKey: "kubernetes.io/hostname",
+							LabelSelector: &metav1.LabelSelector{
+								MatchLabels: podLabels("prometheus", ms.Name),
+							},
+						},
+					},
+					// We cannot expect all clusters to be multi-AZ, especially in the CI.
+					// This is why we set zone-spread as preferred instead of required.
+					PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+						{
+							Weight: 100,
+							PodAffinityTerm: corev1.PodAffinityTerm{
+								LabelSelector: &metav1.LabelSelector{
+									MatchLabels: podLabels("prometheus", ms.Name),
+								},
+								TopologyKey: "topology.kubernetes.io/zone",
+							},
+						},
+					},
+				},
+			},
```

Comment:
Do we want to add this if its not necessarily applicable?
Once we move to SSA we could leave this to the users to add on their own.

Reply:
Wouldn't we want to ensure that Prometheus is run on different nodes so that one can be drained for upgrade while at least one prometheus is running? 
IMHO we should keep this until SSA is ready and remove optional customisations in a different cleanup PR.